lib/rails/graphql/type/object.rb

```diff
@@ -108,6 +109,16 @@ def validate!(*)
             nil # No exception already means valid
           end
 
+          def merge_fields(others)
```

Comment:
Add this function to the `private` section and change to:
```ruby
def merge_fields(*interfaces)
  interfaces.each do |interface|
    interface.fields.each do |name, field|
      if fields.key?(name)
        raise ArgumentError, <<~MSG.squish if fields[name] != field
          The "#{gql_name}" object already have a "#{field.gql_name}" field and it
          it's not equivalent to the one defined on the "#{interface.gql_name}" interface.
        MSG
      else
        fields[name] = field.dup.tap { |x| x.instance_variable_set(:@owner, self) }
      end
    end
  end
end
```

This will account for a situation where 2 interfaces have the same field. If they are compatible, then it's fine, otherwise, it will through this exception.